kolena/_experimental/object_detection/evaluator_multiclass.py

```diff
@@ -195,52 +210,93 @@ def compute_aggregate_label_metrics(
+            unmatched_inf=match_unmatched_inf,
+        )
 
-        tp = [inf for _, inf in match_matched if inf.score >= thresholds[inf.label]]
-        fp = [inf for inf in match_unmatched_inf if inf.score >= thresholds[inf.label]]
-        fn = [gt for gt, _ in match_unmatched_gt] + [
-            gt for gt, inf in match_matched if inf.score < thresholds[inf.label]
-        ]
-        confused = [inf for _, inf in match_unmatched_gt if inf is not None and inf.score >= thresholds[inf.label]]
+        return bbox_matches_for_one_label, samples_count
+
+    def class_metrics_per_test_case(
+        self,
+        label: str,
+        thresholds: Dict[str, float],
+        class_matches: MulticlassInferenceMatches,
+        samples_count: int,
+        average_precision: Optional[float],
+    ) -> ClassMetricsPerTestCase:
+        matched = class_matches.matched
+        unmatched_gt = class_matches.unmatched_gt
+        unmatched_inf = class_matches.unmatched_inf
+
+        tp = [inf for _, inf in matched if inf.score >= thresholds[inf.label]]
+        fp = [inf for inf in unmatched_inf if inf.score >= thresholds[inf.label]]
+        fn = [gt for gt, _ in unmatched_gt] + [gt for gt, inf in matched if inf.score < thresholds[inf.label]]
         tp_count = len(tp)
         fp_count = len(fp)
         fn_count = len(fn)
-        confused_count = len(confused)
         precision = compute_precision(tp_count, fp_count)
         recall = compute_recall(tp_count, fn_count)
         f1_score = compute_f1_score(tp_count, fp_count, fn_count)
-
-        average_precision = 0.0
-        if precision > 0:
-            baseline_pr_curve = compute_pr_curve(all_bbox_matches_for_one_label)
-            if baseline_pr_curve is not None:
-                average_precision = compute_average_precision(baseline_pr_curve.y, baseline_pr_curve.x)
+        if average_precision is None:
+            average_precision = compute_average_precision([precision], [recall])
 
         return ClassMetricsPerTestCase(
             Class=label,
-            nImages=samples_with_label,
+            nImages=samples_count,
             Threshold=thresholds[label],
             Objects=tp_count + fn_count,
             Inferences=tp_count + fp_count,
             TP=tp_count,
             FN=fn_count,
             FP=fp_count,
-            Confused=confused_count,
             Precision=precision,
             Recall=recall,
             F1=f1_score,
             AP=average_precision,
         )
 
+    def compute_aggregate_label_metrics(
+        self,
+        matchings: List[MulticlassInferenceMatches],
+        label: str,
+        thresholds: Dict[str, float],
+    ) -> ClassMetricsPerTestCase:
+        match_and_count: Tuple[MulticlassInferenceMatches, int] = self.bbox_matches_for_one_label(matchings, label)
+        class_matches: MulticlassInferenceMatches = match_and_count[0]
+        samples_count: int = match_and_count[1]
+
+        average_precision: Optional[float] = None
```

Comment:
Is there a reason for computing `average_precision` outside the `class_metrics_per_test_case`?

Reply:
Mainly for bare bones unit testing coverage and cleanliness (avoids including `TestCase` in unit tests, and the evaluator caches).